services/supervisor/src/transactions.rs

```diff
@@ -386,13 +405,21 @@ impl Supervisor {
 
                 ConfigChange::StartService(start_service) => {
                     if !services_to_start.insert(start_service.name.clone()) {
-                        log::trace!(
-                            "Discarded multiple instances with the same name in one request."
-                        );
-                        return Err(Error::MalformedConfigPropose.into());
+                        return Err(Error::MalformedConfigPropose.with_description(
+                            "Discarded multiple instances with the same name in one request.",
+                        ));
                     }
                     start_service.validate(&context)?;
                 }
+
+                ConfigChange::StopService(stop_service) => {
+                    if !modified_instances.insert(stop_service.instance_id) {
```

Comment:
What if `StartService` and `StopService` for the same instance will appear in one config proposal?

Reply:
The `StopService` request will be invalid because the service instance is not running.